hypothesis-python/src/hypothesis/control.py

```diff
@@ -74,18 +74,16 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_value, tb):
         self.assign_variable.__exit__(exc_type, exc_value, tb)
-        if self.close() and exc_type is None:
-            raise CleanupFailed()
-
-    def close(self):
-        any_failed = False
+        errors = []
         for task in self.tasks:
             try:
                 task()
-            except BaseException:
-                any_failed = True
-                report(traceback.format_exc())
-        return any_failed
+            except BaseException as err:
+                errors.append(err)
+        if errors:
+            if len(errors) == 1:
+                raise errors[0] from exc_value
+            raise BaseExceptionGroup("Cleanup failed", errors) from exc_value
```

Comment:
Question: do we need to use `from` with exception groups? I've never tried it :)

Reply:
https://docs.python.org/3/tutorial/errors.html#exception-chaining

It just indicates that we're deliberately replacing the previous exception, instead of hitting another error while handling it.